lib/web_ui/test/engine/platform_views/message_handler_test.dart

```diff
@@ -109,6 +111,64 @@ void testMain() {
               reason:
                   'The response should be a success envelope, with null in it.');
         });
+
+        test('passes creation params to the factory', () async {
+          final List<PlatformViewFactoryCall> factoryCalls = <PlatformViewFactoryCall>[];
+          contentManager.registerFactory(viewType, (int viewId, {Object? params}) {
+            factoryCalls.add((viewId: viewId, params: params));
+            return createDomHTMLDivElement();
+          });
+          final PlatformViewMessageHandler messageHandler = PlatformViewMessageHandler(
+            contentManager: contentManager,
+          );
+
+          final List<Completer<ByteData?>> completers = <Completer<ByteData?>>[];
+
+          completers.add(Completer<ByteData?>());
+          messageHandler.handlePlatformViewCall(
+            _getCreateMessage(viewType, 111),
+            completers.last.complete,
+          );
+
+          completers.add(Completer<ByteData?>());
+          messageHandler.handlePlatformViewCall(
+            _getCreateMessage(viewType, 222, <dynamic, dynamic>{'foo': 'bar'}),
+            completers.last.complete,
+          );
+
+          final List<ByteData?> responses = await Future.wait(
+            completers.map((Completer<ByteData?> c) => c.future),
+          );
+
+          for (final ByteData? response in responses) {
+            expect(
+              codec.decodeEnvelope(response!),
+              isNull,
+              reason: 'The response should be a success envelope, with null in it.',
+            );
+          }
+
+          expect(factoryCalls, hasLength(2));
+          expect(factoryCalls[0].viewId, 111);
+          expect(factoryCalls[0].params, isNull);
+          expect(factoryCalls[1].viewId, 222);
+          expect(factoryCalls[1].params, <dynamic, dynamic>{'foo': 'bar'});
```

Comment:
Do we need an extra case where the param passed is not a map, but a string or an int, or some other simple `Object`?